bootcode/js/src/polyfill-url.js

```diff
@@ -0,0 +1,110 @@
+(function (g) {
+    g.URL = class URL {
+        constructor(url, base) {
+            const pairs = Sidevm.parseURL(url, base);
+            this._pairs = pairs;
+            this.hash = pairs['hash'];
+            this.host = pairs['host'];
+            this.hostname = pairs['hostname'];
+            this.origin = pairs['origin'];
+            this.password = pairs['password'];
+            this.pathname = pairs['pathname'];
+            this.port = pairs['port'];
+            this.protocol = pairs['protocol'];
+            this.search = pairs['search'];
+            this.username = pairs['username'];
+        }
+        get href() {
+            return this.toString();
+        }
+        searchParams() {
+            return new URLSearchParams(this.search);
+        }
+        toString() {
+            let result = this.protocol + '//';
+            if (this.username) {
+                result += this.username;
+                if (this.password) {
+                    result += ':' + this.password;
+                }
+                result += '@';
+            }
+            result += this.hostname;
+            if (this.port) {
+                result += ':' + this.port;
+            }
+            result += this.pathname;
+            if (this.search) {
+                result += this.search;
+            }
+            if (this.hash) {
+                result += this.hash;
+            }
+            return result;
+        }
+
+    }
+    g.URLSearchParams = class URLSearchParams {
```

Comment:
The `URLSearchParams` broken on this case:
```js
const a = new URLSearchParams()
a.append('foo', '1')
a.append('foo', '2')
a.toString() === 'foo=1&foo=2'
```

Reply:
Replaced with core-js.